common/api-review/firestore.api.md

```diff
@@ -252,27 +252,27 @@ export class GeoPoint {
 }
 
 // @public
-export function getCountFromServer(query: Query<unknown>): Promise<AggregateQuerySnapshot<{
+export function getCountFromServer<AppModelType, DbModelType extends DocumentData>(query: Query<AppModelType, DbModelType>): Promise<AggregateQuerySnapshot<{
```

Comment:
There are definitely tradeoffs here as well. But similar to above, we can't use `unknown` for the 2nd type parameter because `unknown` does not extend `DocumentData` and that causes a TypeScript compiler error. We _could_ have used `<unknown, DocumentData>` but I don't personally like the asymmetry of using two different types where both intend to be all-encompassing. We could also have used `<DocumentData, DocumentData>`, which would have forced customers to explicitly drop the type converter on the given query (i.e. `getCountFromServer(query.withConverter(null))`). After discussion, we decided to leave it the way it is in the PR.